carousel/carousel.js

```diff
@@ -0,0 +1,147 @@
+                moveCarouselToRight(carousel, carouselContentHolder);
+            }
+            else if (event.key === "ArrowLeft") {
+                moveCarouselToLeft(carousel, carouselContentHolder);
+            }
+        });
+
+        const carouselContentAndButtons = document.createElement("div");
+        carouselContentAndButtons.classList.add("carousel-content-and-buttons");
+        carouselContentAndButtons.append(carouselLeftButton, carouselContentHolder, carouselRightButton);
+
+        // TODO: Implement generateCarouselDotHolder and connect
+
+        carousel.dataset.currentlyDisplaying = "0";
+        hideAdjacentCarouselItems(carousel, carouselContentHolder);
+        updateLocationOfAdjacentCarouselItems(carousel, carouselContentHolder);
+
+        carousel.addEventListener("transitionend", () => hideAdjacentCarouselItems(carousel, carouselContentHolder));
+
+        carouselAutoplayIntervalMap.set(carousel, setInterval(() => {
+            moveCarouselToRight(carousel, carouselContentHolder);
+        }, AUTOPLAY_TIME_IN_MILLISECONDS));
+
+        carousel.append(carouselTitle, carouselContentAndButtons)
+    }
+}
+
+function generateCarouselContentHolder(carouselContents) {
+    const carouselContentHolder = document.createElement("div");
+    carouselContentHolder.classList.add("carousel-content-holder");
+    carouselContentHolder.append(...carouselContents);
+
+    carouselContents.forEach((carouselItem) => {
+       // TODO
+        carouselItem.classList.add("hidden");
+    });
+
+    return carouselContentHolder;
+}
+
+function generateCarouselDotHolder(amountOfDots) {
+    // TODO: implement
+}
+
+function moveCarouselToRight(carousel, carouselContentHolder) {
+    resetCarouselAutoplay(carousel, carouselContentHolder);
+
+    const carouselChildren = carouselContentHolder.children;
+    carouselChildren[getIncrementOfCurrentlyDisplaying(carousel, carouselContentHolder,1)].classList.remove("hidden");
+
+    // Nested animation frames to assure css animates the transition properly.
+    requestAnimationFrame( () => {
+        updateCurrentlyDisplaying(carousel, carouselContentHolder, 1);
+        requestAnimationFrame(() => {
+            updateLocationOfAdjacentCarouselItems(carousel, carouselContentHolder);
+        });
+    });
+}
+
+function moveCarouselToLeft(carousel, carouselContentHolder) {
+    resetCarouselAutoplay(carousel, carouselContentHolder);
+
+    const carouselChildren = carouselContentHolder.children;
+    carouselChildren[getIncrementOfCurrentlyDisplaying(carousel, carouselContentHolder,-1)].classList.remove("hidden");
+
+    requestAnimationFrame( () => {
+        updateCurrentlyDisplaying(carousel, carouselContentHolder, -1);
+
+        requestAnimationFrame(() => {
+            updateLocationOfAdjacentCarouselItems(carousel, carouselContentHolder);
+        });
+    });
+}
+
+function getIncrementOfCurrentlyDisplaying(carousel, carouselContentHolder, number) {
```

Comment:
This function name is not obvious without going to the code and understand what it means. Maybe getNextSlide?